### PR TITLE
Use mkdocstring syntax for code-api generation

### PR DESCRIPTION
**Pull request by pedro-psb**
The reference doc pages were using sphinx code-gen directives. This replaces those with mkdocstrings directives and tries to apply corresponding configurations where applicable.

To verify it works, you can serve this PR locally and skim through the subtree of:
* `pulpcore/docs/dev/reference/code-api/platform-api/`
* `pulpcore/docs/dev/reference/code-api/plugins-api/`

As noted in the second commit, the introduction of `__all__` is because mkdocstrings [requires this declaration
to display imported members](https://mkdocstrings.github.io/python/usage/configuration/general/?h=import#preload_modules).

### CODE REVIEW

File: pulpcore/plugin/tasking.py [hunk 1]
```diff
@@ -13,3 +13,18 @@
     reclaim_space,
 )
 from pulpcore.app.tasks.repository import add_and_remove  # noqa: F401
```
Comment (mdellweg) [inline pulpcore/plugin/tasking.py:15]: If we do the `__all__` thing. we don't need the `noqa: F401` anymore.

File: docs/dev/reference/code-api/plugins-api/stages.md [hunk 2]
```diff
@@ -9,83 +7,44 @@ to make writing sync code easier. There are several parts to the API:
...
 
 ## Content Related Stages
 
-```{eval-rst}
-.. autoclass:: pulpcore.plugin.stages.ContentSaver
-   :private-members: _pre_save, _post_save
-```
+::: pulpcore.plugin.stages.ContentSaver
+    options:
+       filters: ["!_pre_save", "!_post_save"]
```
Comment (mdellweg) [inline docs/dev/reference/code-api/plugins-api/stages.md:44]: Should we add the filter "!_*" to the mkdoc config?

File: docs/dev/reference/code-api/plugins-api/stages.md [hunk 3]
```diff
@@ -9,83 +7,44 @@ to make writing sync code easier. There are several parts to the API:
 2. The builtin Stages including `artifact-stages` and `content-stages`.
 3. The `stages-api`, which allows you to build custom stages and pipelines.
 
-
-
 ## DeclarativeVersion
```
Comment (mdellweg) [inline docs/dev/reference/code-api/plugins-api/stages.md:10]: This whole file would be much shorter if we just called this section Reference and did `::: pulpcore.plugin.stages`.

**Review by mdellweg — COMMENTED** [review]
Great work so far.

File: docs/dev/reference/code-api/plugins-api/stages.md (hunk 2 above)
Comment (pedro-psb) [reply docs/dev/reference/code-api/plugins-api/stages.md:44]: Yes, probably

File: docs/dev/reference/code-api/plugins-api/stages.md (hunk 3 above)
Comment (pedro-psb) [reply docs/dev/reference/code-api/plugins-api/stages.md:10]: Yeah, but IHO these groupings are helpful.

File: docs/dev/reference/code-api/plugins-api/stages.md (hunk 2 above)
Comment (pedro-psb) [reply docs/dev/reference/code-api/plugins-api/stages.md]: https://github.com/pulp/pulp-docs/pull/91

**Pull request merged**